girder/models/file.py

```diff
@@ -309,6 +323,10 @@ def createFile(self, creator, item, name, size, assetstore, mimeType=None,
             'itemId': item['_id'] if item else None
         }
 
+        if assetstoreModel:
+            file['assetstoreModel'] = assetstoreModel
+            file['assetstoreModelPlugin'] = assetstoreModelPlugin
```

Comment:
Does it make sense for this metadata to be assigned to files even if they don't have a non-standard assetstoreModel/assetstoreModelPlugin ?

Reply:
Also,  if we just passed in the class (or the model object) rather than a string,  couldn't we enforce an API on assetstore models that returns this data.  That way createFile only has to query the assetstore model object (or class or whatever)  for this information?  

maybe this is going to far,  but if the assetstore argument was an actual object (instead of a dict)  then all of this could be pulled off of that object right?

Reply:
> Does it make sense for this metadata to be assigned to files even if they don't have a non-standard assetstoreModel/assetstoreModelPlugin ?

We could do that, though preserving the old mode also ensures that legacy files will work without any need for data migration.

> if the assetstore argument was an actual object (instead of a dict) then all of this could be pulled off of that object right?

Not at this time, no. The registration of models to a string is currently a one-way pointer. The model itself doesn't know under what string it's registered, though that might be a nice future enhancement.

Reply:
i guess there have been some conversations about schema migration.  Don't let my concerns hold things up.  I was just curious as I was looking through.  I assume you want @manthey to make sure this is going to work for the database assetstore downstream right?

Reply:
Yep, that's correct.